lib/temp-directory.js

```diff
@@ -2,21 +2,38 @@
 var path = require('path');
 
 // npm modules
+var async = require('async');
 var mkdirp = require('mkdirp');
 var rimraf = require('rimraf');
 var uuid = require('uuid');
 var osenv = require('osenv');
 
 function create(context, next) {
   var _path = path.join(osenv.tmpdir(),uuid.v4());
-  var npmConfigTmp = path.join(_path, 'npm_config_tmp');
-
-  context.path = _path;
+  var npmConfigTmp = _path;
+  // Overwrite the npm temp directory if not already set via the env variable
+  if (!process.env.npm_config_tmp) {
+    // Use the same directory as the citgm temp directory for npm if set
+    if (context.options && context.options.tmpDir) {
+      npmConfigTmp = path.join(context.options.tmpDir, 'npm_config_tmp');
+    } else {
+      // Else use the system temp directory
+      npmConfigTmp = path.join(_path, 'npm_config_tmp');
+    }
+  } else {
+    // Use the env var's temp directory if set
+    npmConfigTmp = process.env.npm_config_tmp;
+  }
   context.npmConfigTmp = npmConfigTmp;
 
-  context.emit('data', 'silly', context.module.name + ' mk.tempdir', _path);
+  if (context.options && context.options.tmpDir) {
+    context.path = path.join(context.options.tmpDir, context.module.name);
+  } else {
+    context.path = npmConfigTmp;
```

Comment:
why would we set the context.path to npmConfigTmp? we want it to be _path, not the same directory